src/models/vac.ts

```diff
@@ -9,14 +9,14 @@ import * as sddInformations from "../static/SDD.mapping.json";
  */
 export const genVaccinationRecord = (vaccinationRecord: VaccinationRecord, expiryDateTime: string, issuerName: string, reference: string): VaccineRecord[] => {
 
-  /* assign primary series of vaccination records by filtering `vaccinationDateTime` is before `effectiveDate` */
+  /* assign primary series of vaccination records by filtering `vaccinationDateTime` is before `effectiveDate` or without `effectiveDate` */
   const primaryVaccinations: Vaccination[] = (vaccinationRecord.vaccinations).filter((vaccination) =>
-    isBefore(parseISO(vaccination.vaccinationDateTime), parseISO(vaccinationRecord.effectiveDate))
+    vaccinationRecord.effectiveDate? isBefore(parseISO(vaccination.vaccinationDateTime), parseISO(vaccinationRecord.effectiveDate)): true
   );
 
-  /* set`isBooster` flag in vaccination records when `vaccinationDateTime` is after CMB `effectiveDate` */
+  /* set`isBooster` flag as `true` in vaccination records when `vaccinationDateTime` is after CMB `effectiveDate` */
   const euVaccinations: Vaccination[] = (vaccinationRecord.vaccinations).map((vaccination) => {
-    vaccination.isBooster = isAfter(parseISO(vaccination.vaccinationDateTime), parseISO(vaccinationRecord.effectiveDate));
+    vaccination.isBooster = vaccinationRecord.effectiveDate? isAfter(parseISO(vaccination.vaccinationDateTime), parseISO(vaccinationRecord.effectiveDate)): false;
```

Comment:
How about this logic? If no `effectiveDate`, it will be deemed as booster too?

Cause it seems as though without `effectiveDate`, it is both considered primary and booster?

Reply:
it's for setting the booster flag in each vaccinations record. If not effectiveDate, set as false cos it's not booster.